docs/source/conf.py

```diff
@@ -313,6 +313,7 @@
         "https://asyncssh.readthedocs.io/en/latest/",
         "https://asyncssh.readthedocs.io/en/latest/objects.inv",
     ),
+    "pyarrow": ("https://arrow.apache.org/docs/python/", None),
```

Comment:
Dropping `python/` at the end seems to work for me locally. Could you give this a try?

```suggestion
    "pyarrow": ("https://arrow.apache.org/docs/", None),
```

Reply:
Thanks. Also worked for me locally